php/includes/language.php

```diff
@@ -91,3 +91,40 @@ function FN_(string $singular, string $plural, ...$replacePairs)
 		wrap_array_keys_in_braces($replacePairs),
 	);
 }
+
+// custom translation framework
+define(
+	"TRANSLATION_TABLE",
+	array(
+		"kHTML_LANGUAGE" => array(
+				/* The 'lang' attribute for <html> tag */
+				"en" => "en",
+				"fr" => "fr",
+		),
+		"kA_MARIO_KART_GAME_FOR_BROWSER" => array(
```

Comment:
I used some kind of constants here. However, it may be more intuitive to use the english translation of the strings as we used to do with gettext, as it provides a reasonable default value.
A potential issue, however, is that some strings are huge (e.g. some <p> with the history of mkpc) while others could use problematic symbols…

Reply:
I like the idea of having translation keys yeah, this way you don't repeat yourself and the "translation file" is cleaner